node_modules/oae-tenants/lib/api.js

```diff
@@ -289,6 +288,52 @@ var createTenant = module.exports.createTenant = function(ctx, alias, displayNam
 };
 
 /**
+ * Internal function that creates and persists a tenant. It's assumed
+ * that the tenant parameters have been validated previously.
+ *
+ * @param  {String}     alias                   The unique alias for the tenant
+ * @param  {String}     displayName             A descriptive short name for the tenant
+ * @param  {String}     host                    The host on which this tenant will be proxying (e.g. oae.cam.ac.uk or oae.gatech.edu)
+ * @param  {Function}   callback                Standard callback function
+ * @param  {Object}     callback.err            An error that occurred, if any
+ * @param  {Tenant}     callback.tenant         The created tenant
+ * @api private
+ */
+var _createTenant = function(alias, displayName, host, callback) {
+    // Create the tenant
+    var tenant = new Tenant(alias, displayName, host);
+    Cassandra.runQuery('UPDATE "Tenant" SET "displayName" = ?, "host" = ?, "active" = ? WHERE "alias" = ?',
+        [tenant.displayName, host, tenant.active, tenant.alias], function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // This event is not strictly necessary as it will be emitted by our PubSub publisher
+        // as well. We emit it before we return to the caller so our unit tests can keep track
+        // of config updates BEFORE the REST request completes. If we didn't there would be a
+        // short period between the REST request returning and the config re-caching where the
+        // config would be out-of-date. The length of this period is determined by how fast Redis
+        // can broadcast the pubsub messages. When the system is under load and suffering from IO
+        // starvation (such as during unit tests) this could lead to intermittent test failures.
+        // The downside of emitting this event here is that it will lead to authentication strategies
+        // and config elements being recached twice
+        TenantsAPI.emit('created', tenant);
```

Comment:
I'm not exactly sure why emitting this event here will result in config elements and authentication strategies being recached twice?

Also, maybe rather than emitting a `created` event, we just do a lazy-load. It's only `update` that really needs a pubsub message.